Doc/reference/datamodel.rst

```diff
@@ -1395,12 +1395,14 @@ Basic customization
    context (e.g., in the condition of an ``if`` statement), Python will call
    :func:`bool` on the value to determine if the result is true or false.
```

Comment:
"Python will call bool..." is not true if value is NotImplemented.

As the new text says, a.__eq__(b) and b.__eq__(a) return NotImplemented when a is not b.  In this case,  and then a == b is False even though bool(NotImplemented) is True, and Deprecated.  I presume the == logic directly interprets NotImplemented.  But I don't know if such special casing is always true, as it would not be needed for != and this is a different issue.  (In fact, I noticed it *because* of the new text.)

Reply:
Since I triggered the bot by approving, I went ahead and did the same for the backports.

Reply:
It's a bit more nuanced. At the opcode level, if `NotImplemented` is returned for anything but `==` or `!=`, then `TypeError` is raised. And for `==` and `!=`, they fall back to `is` and `is not`, respectively. So the statement still holds, albeit by glossing over _why_ it holds. (see https://snarky.ca/unravelling-rich-comparison-operators/ for details.)